### PR TITLE
Refactor dashboard view to class-based

### PR DESCRIPTION
**Pull request by RMadjev**

### CODE REVIEW

File: tcms/core/views.py [hunk 1]
```diff
@@ -7,47 +7,50 @@
...
-        'test_plans_disable_count': test_plans_disable_count,
-        'last_15_test_plans': test_plans.filter(is_active=True)[:15],
-
-        'last_15_test_runs': test_runs[:15],
-
-        'test_runs_count': test_runs.count(),
-    }
-    return render(request, 'dashboard.html', context_data)
+@method_decorator(login_required, name='dispatch')
+class DashboardView(TemplateView):
```
Comment (atodorov) [inline tcms/core/views.py]: missing pylint disable,
missing method names

File: tcms/core/views.py [hunk 2]
```diff
@@ -7,47 +7,50 @@
...
+        ).order_by('-run_id').distinct()
+
+        return {
+            'test_plans_count': test_plans.count(),
+            'test_plans_disable_count': test_plans_disable_count,
+            'last_15_test_plans': test_plans.filter(is_active=True)[:15],
+
+            'last_15_test_runs': test_runs[:15],
+
+            'test_runs_count': test_runs.count(),
```
Comment (atodorov) [inline tcms/core/views.py]: drop the empty lines between dict entries

File: tcms/core/views.py (hunk 1 above)
Comment (RMadjev) [reply tcms/core/views.py]: Its a TemplateView, and the method names are not necessary in my opinion. It implements only get method. By itself TemplateView shows that only get is expected.

Comment (atodorov) [reply tcms/core/views.py]: FTR specifying the allowed methods in `http_method_names` short-circuits the dispatch logic before returning a 405 response. We can merge this as-is ATM.

**Pull request merged**